compiler/rustc_hir_analysis/src/astconv/errors.rs

```diff
@@ -585,6 +586,35 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
             }
         }
 
+        // We get all the associated items that _are_ set,
+        // so that we can check if any of their names match one of the ones we are missing.
+        // This would mean that they are shadowing the associated type we are missing,
+        // and we can then use their span to indicate this to the user.
+        let bound_names = trait_bounds
+            .iter()
+            .filter_map(|poly_trait_ref| {
+                poly_trait_ref.trait_ref.path.segments.last().and_then(|path| {
+                    let args = path.args?;
+                    Some((path, args))
+                })
+            })
+            .flat_map(|(path, args)| {
+                args.bindings.iter().map(|binding| {
+                    let ident = binding.ident;
+                    let trait_def = path.res.opt_def_id();
+                    let assoc_item = trait_def.and_then(|did| {
+                        tcx.associated_items(did).find_by_name_and_kinds(
+                            tcx,
+                            ident,
+                            &[AssocKind::Fn, AssocKind::Type, AssocKind::Const],
```

Comment:
```suggestion
                            &[AssocKind::Type],
```
Only types can be used in the `Trait<X = Y>` constructions.